command/sync.go

```diff
@@ -547,7 +547,7 @@ func generateDestinationURL(srcurl, dsturl *url.URL, isBatch bool) *url.URL {
 }
 
 // shouldSkipObject checks is object should be skipped.
-func (s Sync) shouldSkipObject(object *storage.Object, verbose bool) bool {
+func (s Sync) shouldSkipObject(object *storage.Object, isSrc, verbose bool) bool {
```

Comment:
lets try to separate this into two different functions:
- shouldSkipSourceObject
- shouldSkipDestObject

we also need to check the existence of `--delete` flag when deciding whether to skip or not.